hls4ml/converters/onnx_to_hls.py

```diff
@@ -125,252 +183,95 @@ def onnx_to_hls(yamlConfig):
-    'LeakyRelu':'LeakyReLU', 'ThresholdedRelu':'ThresholdedReLU', 'HardSigmoid':'Activation',
-    'Elu':'ELU', 'Selu':'Activation', 'PRelu':'PReLU', 'Softmax':'Softmax', 'Softsign':'Activation', 'Softplus':'Activation',
-    'Sum':'Add', 'Sub':'Subtract', 'Max':'Maximum', 'Min':'Minimum', 'Mul':'Multiply', 'Concat':'Concatenate'}
+    #Optimizie the model graph's before conversion
+    model = shape_inference.infer_shapes(model) # have to infer shapes before optimizing the model
+    graph =  _hls4ml_onnx_optimizer(model.graph)
     
-    #Define layers to skip for conversion to HLS
-    skip_layers = ['Squeeze', 'Unsqueeze', 'Dropout', 'Identity', 'Flatten', 'Transpose', 'Reshape'] 
     #Map inputs of skipped layers
     inputs_map = {}
-
-    passes = ['fuse_transpose_into_gemm', 'fuse_matmul_add_bias_into_gemm', 'eliminate_nop_transpose', 'fuse_consecutive_transposes']
-    model = shape_inference.infer_shapes(model) # have to infer shapes before optimizing the model
-    model = optimizer.optimize(model, passes)
-    model = shape_inference.infer_shapes(model) # have to infer shapes before optimizing the model
-    
     reader = ONNXDataReader(model)
+    
+    #Input layer info (assuming that there is only one input layer)
+    input_layer = {}
+    input_layer['name'] = 'Input'
+    input_layer['class_name'] = 'InputLayer'
+    input_shape = [d.dim_value for d in graph.input[0].type.tensor_type.shape.dim]
+    input_layer['input_shape'] = input_shape
 
-    #Loop through layers
-    layer_counter = 0
-    all_inputs = [x.name for x in model.graph.input]
-    all_initializers = [x.name for x in model.graph.initializer]
-    input_layers = [x for x in all_inputs if x not in all_initializers]
-    output_layers = [x.name for x in model.graph.output]
-
-    for i, inp in enumerate(input_layers):
-        input_layer = {}
-        input_layer['name'] = inp
-        input_layer['class_name'] = 'InputLayer'
-        inp_shape = next((x.type.tensor_type.shape.dim for x in model.graph.input if x.name == inp), None)
-        input_layer['input_shape'] = [x.dim_value for x in inp_shape]
-        if len(input_layer['input_shape']) > 1:
-            input_layer['input_shape'][0] = None
-
-        input_layer['outputs'] = [inp]
+    if len(input_layer['input_shape']) > 1:
+        input_layer['input_shape'][0] = None #Firt dim is batch
 
-        sanitize_layer_name(input_layer)
-        input_layers[i] = input_layer['name']
-        layer_list.append(input_layer)
+    sanitize_layer_name(input_layer)
+    layer_list.append(input_layer)
 
-    # Check for unsupported layer type
-    for operation in model.graph.node:
-        if operation.op_type not in supported_operations:
-            raise Exception('ERROR: Unsupported operation type: {}'.format(operation.op_type))
+    # Defined supported layers and check for unsupported layer type
+    skip_layers = ['Squeeze', 'Unsqueeze', 'Dropout', 'Identity', 'Flatten', 'Reshape']
```

Comment:
Reshape should be handled, we have models that need to handle it, more will come in the future. Squeeze/Unsqueeze can probably be folded into Reshape as well.